pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -3066,6 +3036,19 @@ public void onUpdate(TopicPolicies policies) {
         });
     }
 
+    private CompletableFuture<Void> updateSubscriptionsDispatcherRateLimiter() {
+        List<CompletableFuture<Void>> consumerCheckFutures = new ArrayList<>();
+        subscriptions.forEach((subName, sub) -> sub.getConsumers().forEach(consumer -> {
+            consumerCheckFutures.add(consumer.checkPermissionsAsync().thenRun(() -> {
+                Dispatcher dispatcher = sub.getDispatcher();
+                if (dispatcher != null) {
+                    dispatcher.updateRateLimiter();
+                }
+            }));
+        }));
+        return CompletableFuture.allOf(consumerCheckFutures.toArray(new CompletableFuture[0]));
```

Comment:
Is this different with `FutureUtil.waitForAll(consumerCheckFutures)`?

Reply:
They are same. `FutureUtil.waitForAll(consumerCheckFutures)` make less modification for code. I will update it